src/test/java/api/SalesforceApi.java

```
@@ -0,0 +1,92 @@
+import pageObjects.BasePage;
+import pageObjects.FakerData;
+
+public class SalesforceApi extends BasePage {
+
+FakerData fakerData = new FakerData();
+double randNumber = Math.random() * 10000 + 1;
+private String leadSource = "WeWork.com" + randNumber;
+private static HttpURLConnection con;
+private static final String TOKEN_URL =  ENV.get("TOKEN_URL");
+private String firstName = fakerData.getFirstName();
+private String lastName = fakerData.getLastName();
+private String email = fakerData.getEmail();
+private String company = fakerData.getCompnay();
+private String address = fakerData.getStreetAddress();
+
+    public String customerStr() {
+        System.out.println(fakerData.getFirstName());
+        System.out.println(fakerData.getLastName());
+        System.out.println(leadSource);
+        return fakerData.getFirstName() + " " + fakerData.getLastName() +  "-Tel Aviv-Yafo-" + leadSource;
+    }
+
+    HttpClient httpclient = new DefaultHttpClient();
+
+    private String generateAccessToken() throws IOException {
+        HttpPost post = new HttpPost("https://test.salesforce.com/services/oauth2/token");
+        List<NameValuePair> params = new ArrayList<NameValuePair>();
+        params.add(new BasicNameValuePair("client_id", ENV.get("CLIENT_ID")));
+        params.add(new BasicNameValuePair("client_secret", ENV.get("CLIENT_SECRET")));
+        params.add(new BasicNameValuePair("grant_type", "password"));
+        params.add(new BasicNameValuePair("username", ENV.get("USERNAME")));
+        params.add(new BasicNameValuePair("password", ENV.get("PASSWORD")));
+        params.add(new BasicNameValuePair("securityToken", "undefined"));
+        post.setEntity(new UrlEncodedFormEntity(params));
+        HttpResponse response = httpclient.execute(post);
+        String body = EntityUtils.toString(response.getEntity());
+        System.out.println("OAuth" + body);
+        JSONObject obj = new JSONObject(body);
+        return obj.getString("access_token");
```

What happens if the call failed for some reason and access_token wasn't returned? Will this throw an exception or will it return null? If throwing an exception, that's good. If we return null, we probably want to check for that case first so that we can throw a meaningful error message and help the person debugging the failure.

---

replaced the IOException with Exception